phoenix-core/src/test/java/org/apache/phoenix/coprocessor/MetaDataEndpointImplTest.java

```diff
@@ -0,0 +1,299 @@
+package org.apache.phoenix.coprocessor;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.phoenix.end2end.ParallelStatsDisabledIT;
+import org.apache.phoenix.exception.SQLExceptionCode;
+import org.apache.phoenix.jdbc.PhoenixDatabaseMetaData;
+import org.apache.phoenix.schema.PColumn;
+import org.apache.phoenix.schema.PTable;
+import org.apache.phoenix.schema.TableNotFoundException;
+import org.apache.phoenix.util.PhoenixRuntime;
+import org.junit.Test;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+public class MetaDataEndpointImplTest extends ParallelStatsDisabledIT {
```

Comment:
One test that would be useful (and maybe you already have it?) is to create a table, create a view (maybe on a different RS). Then drop the table and recreate it and the view with the same name but different columns. Make sure that the lazy cleanup code cleaned up the left over state correctly.

Reply:
I removed the OrphanCleaner code that was there previously and I create a PHOENIX-3534 to handle cleanup of child view metadata for a parent that was dropped during compaction. Instead of having additional clean up code, I think we should just detect that we are trying to re-create a table that was dropped whose child view metadata wasn't cleaned up and then throw an exception.

Reply:
I think we should keep the OrphanCleaner code. The idea is that failed deletions of metadata are transparent. They shouldn't block creation of new tables. We should also cleanup on compaction.

Reply:
@JamesRTaylor Can you please review? I modified the PR to drop child metadata when we try to reuse a table name of a table that was dropped. It also handles reusing a view name / view index name. 
I also added a config that determines whether or not SYSTEM.CATALOG can split. I merged the latest changes from master as well.